### ISSUE TITLE
Analyzer suggestion: avoid incremental string concatenation in loops

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn-analyzers#5914]
**Languages applicable:**
Both C# and VB

**Code example that the analyzer should report:**
```
public string ExampleMethod(int num)
{
    string str = string.Empty;

    for (int i = 0; i < num; i++)
    {
        [|str|] += i;
    }

    return str;
}
```

**Message**: avoid incremental string concatenation in loops

**Additional information:**
I am not sure, if "incremental concatenation" is the right term. So a bit of clarification: by "incremental concatenation" I mean situation, when appending to an existing string, like `str += anotherStr` or `str = str + anotherStr`. So this code should not produce described diagnostics:
```
public string[] ExampleMethod(string baseStr, int num)
{
    string[] result = new string[num];

    for (int i = 0; i < num; i++)
    {
        result[i] = baseStr + num; // No diagnostics here
    }

    return result;
}
```

**Possible codefix:**
Replace string concatenation with `StringBuilder`:
```
public string ExampleMethod(int num)
{
    StringBuilder strBuilder = new(); // this line shold depend on current .editorconfig or IDE settings.
    // For instance, another possible variant of creating string builder is "var strBuilder = new StringBuilder();" 

    for (int i = 0; i < num; i++)
    {
        strBuilder.Append(num);
    }

    return strBuilder.ToString();
}
```
